File: Data Structures & Algorithms/add-two-numbers/submission-0.js
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     constructor(val = 0, next = null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

class Solution {
    /**
     * @param {ListNode} l1
     * @param {ListNode} l2
     * @return {ListNode}
     */
    addTwoNumbers(l1, l2) {
        let v1,v2,sum,result=new ListNode(),current=result,carry=0
        while(l1 || l2 || carry){
            v1=v2=0
            if(l1){
                v1=l1.val
                l1=l1.next
            }
            if(l2){
                v2=l2.val
                l2=l2.next
            }
            sum=v1+v2+carry
            if(sum>9){
                carry=Math.floor(sum/10)
                sum=sum%10
            }else{
                carry=0
            }
            current.next=new ListNode(sum,null)
            current=current.next
        }
        return result.next
    }
}

File: Data Structures & Algorithms/merge-k-sorted-linked-lists/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:    
    def mergeKLists(self, lists: List[Optional[ListNode]]) -> Optional[ListNode]:
        while len(lists)>1:
          mergedList=[]
          for i in range(0,len(lists),2):
            mergedList.append(self.mergeList(lists[i],lists[i+1] if i+1<len(lists) else None))
          lists=mergedList
        return lists[0] if lists else None
    def mergeList(self,list1,list2):
      if not list1:
        return list2
      if not list2:
        return list1
      dummy=ListNode()
      current=dummy
      while list1 and list2:
        if list1.val<=list2.val:
          current.next=list1
          list1=list1.next
        else:
          current.next=list2
          list2=list2.next
        current=current.next
      if list1:
        current.next=list1
      elif list2:
        current.next=list2
      return dummy.next